generators/php/text.js

```diff
@@ -0,0 +1,254 @@
+  var where2 = block.getFieldValue('WHERE2');
+  var at1 = Blockly.PHP.valueToCode(block, 'AT1',
+      Blockly.PHP.ORDER_FUNCTION_CALL) || '0';
+  var at2 = Blockly.PHP.valueToCode(block, 'AT2',
+      Blockly.PHP.ORDER_FUNCTION_CALL) || '0';
+  if (where1 == 'FIRST' && where2 == 'LAST') {
+    var code = text;
+  } else {
+    var functionName = Blockly.PHP.provideFunction_(
+        'text_get_substring',
+        [ 'function ' + Blockly.PHP.FUNCTION_NAME_PLACEHOLDER_ +
+            '($text, $where1, $at1, $where2, $at2) {',
+          '    if ($where2 == \'FROM_START\') {',
+          '      $at2--;',
+          '    } else if ($where2 == \'FROM_END\') {',
+          '      $at2 = $at2 - $at1;',
+          '    } else if ($where2 == \'FIRST\') {',
+          '      $at2 = 0;',
+          '    } else if ($where2 == \'LAST\') {',
+          '      $at2 = strlen($text);',
+          '    } else { $at2 = 0; }',
+          '    if ($where1 == \'FROM_START\') {',
+          '      $at1--;',
+          '    } else if ($where1 == \'FROM_END\') {',
+          '      $at1 = strlen($text) - $at1;',
+          '    } else if ($where1 == \'FIRST\') {',
+          '      $at1 = 0;',
+          '    } else if ($where1 == \'LAST\') {',
+          '      $at1 = strlen($text) - 1;',
+          '    } else { $at1 = 0; }',
+          '  return substr($text, $at1, $at2);',
+          '}']);
+    var code = functionName + '(' + text + ', \'' +
+        where1 + '\', ' + at1 + ', \'' + where2 + '\', ' + at2 + ')';
+  }
+  return [code, Blockly.PHP.ORDER_FUNCTION_CALL];
+};
+
+Blockly.PHP['text_changeCase'] = function(block) {
+  // Change capitalization.
+  var code;
+  if (block.getFieldValue('CASE')=='UPPERCASE') {
+    var argument0 = Blockly.PHP.valueToCode(block, 'TEXT',
+            Blockly.PHP.ORDER_FUNCTION_CALL) || '\'\'';
+    code = 'strtoupper(' + argument0 + ')';
+  } else if (block.getFieldValue('CASE')=='LOWERCASE') {
+    var argument0 = Blockly.PHP.valueToCode(block, 'TEXT',
+            Blockly.PHP.ORDER_FUNCTION_CALL) || '\'\'';
+    code = 'strtolower(' + argument0 + ')';
+  } else if (block.getFieldValue('CASE')=='TITLECASE') {
+    var argument0 = Blockly.PHP.valueToCode(block, 'TEXT',
+            Blockly.PHP.ORDER_FUNCTION_CALL) || '\'\'';
+    code = 'ucwords(strtolower(' + argument0 + '))';
+  }
+  return [code, Blockly.PHP.ORDER_FUNCTION_CALL];
+};
+
+Blockly.PHP['text_trim'] = function(block) {
+  // Trim spaces.
+  var OPERATORS = {
+    'LEFT': 'ltrim',
+    'RIGHT': 'rtrim',
+    'BOTH': 'trim'
+  };
+  var operator = OPERATORS[block.getFieldValue('MODE')];
+  var argument0 = Blockly.PHP.valueToCode(block, 'TEXT',
+      Blockly.PHP.ORDER_ATOMIC) || '\'\'';
+  return [ operator + '(' + argument0 + ')', Blockly.PHP.ORDER_ATOMIC];
+};
+
+Blockly.PHP['text_print'] = function(block) {
+  // Print statement.
+  var argument0 = Blockly.PHP.valueToCode(block, 'TEXT',
+      Blockly.PHP.ORDER_FUNCTION_CALL) || '\'\'';
```

Comment:
ORDER_FUNCTION_CALL isn't correct here, the code is going inside print's parentheses.  This should be ORDER_NONE.

The result can be seen in this generated code:
print((count($list2)));